spec/controllers/wishlist_controller_spec.rb

```diff
@@ -0,0 +1,158 @@
+require 'rails_helper'
+
+RSpec.describe WishlistController, type: :controller do
+  let(:cas_session) { {cas_user: 'bob'} }
+  let(:wishlist_item) { double('wishlist_item') }
+  let(:relation) { double(first_or_initialize: wishlist_item) }
+
+  describe 'POST #add_course' do
+    context 'when the item can be saved' do
+      before { allow(wishlist_item).to receive(:save).and_return true }
```

Comment:
Yeah, maybe some more detail for posterity:

`allow(thing).to receive(:method).and_return value ` stubs the method on thing, so that when it is called it returns the specified value instead of calling the actual method on the object. There a whole bunch of other cool things that you can do with stubbing, which you can read in the docs :). It's used here because it's a really great way to make saving objects fail without having to seed them with data that will fail validations (logic which would need to be updated any time the validations on that object change).